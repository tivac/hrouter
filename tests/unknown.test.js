"use strict";

const router = require("../src/router.js");

describe("router", () => {
    const original = router._unknown;
    
    beforeEach(() => {
        router._routes = Object.create(null);

        router._unknown = original;
    });

    describe("router.unknown", () => {
        it("should throw an error by default", () => {
            router("/fooga", jest.fn());

            expect(() => router.go("/wooga")).toThrowErrorMatchingSnapshot();
        });

        it("should allow providing a new unknown handler", () => {
            const fn = jest.fn();
            
            router.unknown(fn);

            router("/fooga", jest.fn());

            router.go("/wooga");

            expect(fn.mock.calls).toMatchSnapshot();
        });
    });
});
