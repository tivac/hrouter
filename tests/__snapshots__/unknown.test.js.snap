// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`router router.unknown should allow providing a new unknown handler 1`] = `
Array [
  Array [
    Object {
      "url": "/wooga",
    },
  ],
]
`;

exports[`router router.unknown should throw an error by default 1`] = `"Unknown path: /wooga"`;
